vtu11/impl/parallel_helper_impl.hpp

```diff
@@ -0,0 +1,69 @@
+﻿//          __        ____ ____
+// ___  ___/  |_ __ _/_   /_   |
+// \  \/ /\   __\  |  \   ||   |
+//  \   /  |  | |  |  /   ||   |
+//   \_/   |__| |____/|___||___|
+//
+//  License: BSD License ; see LICENSE
+//
+#ifndef VTU11_PARALLEL_HELPER_IMPL_HPP
+#define VTU11_PARALLEL_HELPER_IMPL_HPP
+
+#include "vtu11_impl.hpp"
+namespace vtu11
+{
+  template<typename Writer>
+  void writePVTUfile( const std::string& path,
+                      std::string baseName,
+                      const std::vector<DataSet>& pointData,
+                      const std::vector<DataSet>& cellData,
+                      size_t numberOfFiles,
+                      Writer writer)
+  {
+    std::string parallelName = path + baseName + ".pvtu";
+    std::ofstream output( parallelName, std::ios::binary );    
+    size_t ghostLevel = 0;//Hardcoded to be 0
+    std::vector<double> points;
+    VTU11_CHECK( output.is_open( ), "Failed to open file \"" + baseName + "\"" );
+
+    output << "<?xml version=\"1.0\"?>\n";
+    StringStringMap headerAttributes{ { "byte_order",  endianness()       },
+                                      { "type"      ,  "PUnstructuredGrid" },
+                                      { "version"   ,  "0.1"              } };
+
+    writer.addHeaderAttributes( headerAttributes );
+    {
+      ScopedXmlTag vtkFileTag( output, "VTKFile", headerAttributes );
+      {
+        ScopedXmlTag pUnstructuredGridFileTag( output, "PUnstructuredGrid", { { "GhostLevel", std::to_string( ghostLevel ) } });
+        {
+          ScopedXmlTag pPointDataTag( output, "PPointData", { } );
+
+          for( const auto& dataSet : pointData )
+          {
+            detail::addDataSet( writer, output, std::get<2>( dataSet ), std::get<1>( dataSet ), std::get<0>( dataSet ), true);
+          }
+        } // PPointData
+        {
+          ScopedXmlTag pCellDataTag( output, "PCellData", { } );
+
+          for( const auto& dataSet : cellData )
+          {
+            detail::addDataSet( writer, output, std::get<2>( dataSet ), std::get<1>( dataSet ), std::get<0>( dataSet ), true );
+          }
+        } // PCellData
+        {
+          ScopedXmlTag pPointsTag( output, "PPoints", {} );
+          detail::addDataSet( writer, output, points, 3, "", true );
+        } // PPoints
+        for( size_t nFiles = 0; nFiles < numberOfFiles; ++nFiles )
+        {
+          std::string pieceName = baseName + "/" + baseName + "_" + std::to_string( nFiles ) + ".vtu";
+          writeEmptyTag( output, "Piece", { { "Source", pieceName } } );
+        } // Pieces
+      } // PUnstructuredGrid
+    } // VTKFile
```

Comment:
seems to be a wrong comment

Reply:
Not sure, but it looks correct to me. For example there is a scope in which PUnstructuredGrid is defined. At the end of that scope the desctructor of PUnstructuredGrid is called, which writes the closing xml tag into the file. 

That's why the comment is at the end of the enclosing scope.